code/iaas/allocator/src/main/java/io/cattle/platform/allocator/constraint/VolumeAccessModeConstraintProvider.java

```diff
@@ -34,10 +35,10 @@ public void appendConstraints(AllocationAttempt attempt, AllocationLog log, List
             List<Volume> volumes = InstanceHelpers.extractVolumesFromMounts(instance, objectManager);
             for (Volume v : volumes) {
                 if (VolumeConstants.ACCESS_MODE_SINGLE_HOST_RW.equals(v.getAccessMode())) {
-                    if (v.getHostId() != null) {
+                    Long hostID = DataAccessor.fieldLong(v, VolumeConstants.FIELD_LAST_ALLOCATED_HOST_ID);
+                    if (hostID != null ) {
```

Comment:
Remove this check, right? We still want to add this constraint if hostID == null. It would be solely based off of hostIds returned by findHostsWithVolumeInUse

Though, we'll still use v.hostId as a fallback if their are no mounts and furthermore if there are no mounts and v.hostId is null, we dont add the constraint at all.